app/models/reviewable_flagged_post.rb

```diff
@@ -0,0 +1,284 @@
+require_dependency 'reviewable'
+
+class ReviewableFlaggedPost < Reviewable
+
+  def post
+    @post ||= (target || Post.with_deleted.find_by(id: target_id))
+  end
+
+  def build_actions(actions, guardian, args)
```

Comment:
Is there any chance we could move action creation to: `lib/reviewable/actions.rb`.It's not directly related to the actions :thinking:

Reply:
Can you elaborate? The actions are built dynamically per reviewable type, so these actions are `ReviewableFlaggedPost` specific. You'd have to have `reviewable/actions.rb` test the type of object in some kind of case statement I think.

Reply:
This is a pretty complex class that not only perform the review actions but also decides which actions will be visible in the UI. I was asking if we could move some of that complexity away by moving that responsibility to `actions.rb` (if it makes sense)

Reply:
The actions are used for display in the UI, but it's also used for access checks. For example, if a reviewable is in the `approved` state, it won't return an `approve` action, and if an API request comes in that attempts to review it, it will fail because that action does not exist.

The problem with moving it to actions is then the actions class needs to know about all types of reviewables and how they should respond to things so it could build the appropriate actions.

Reply:
That's why I bring this up. This class is taken care of:
  - UI elements
  - Authorization
  - Persistence

and is also an entity of our business domain.

My main intention is just to raise awareness that tight coupling leads to systems that are harder to extend and evolve :stuck_out_tongue_winking_eye:

Reply:
I think in this case I'm still seeing how you would improve things. If you like, you could create a follow up PR showing how you'd split things up and how it would improve the code.